ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs

```diff
@@ -58,6 +58,11 @@ import           Ouroboros.Consensus.Util.IOLike
 --   of transactions that fits in a block.
 -- * It supports wallets that submit dependent transactions (where later
 --   transaction depends on outputs from earlier ones).
+--
+-- When only one thread is operating on the mempool, operations that mutate the
+-- state based on the input arguments (tryAddTxs and removeTxs) will produce the
+-- same result whether they process transactions one by one or all in one go,
+-- i.e.  those operations are "distributive".
```

Comment:
I don't think `distributive` is the right choice. Here's what we have:

`add [x,y] ~ add [x] >> add [y]`

That phrasing makes it rather clear that `add` is a homomorphism from `++` to `>>`. That wording is very precise, but is admittedly some heavyweight jargon. Hopefully including the code snippet I wrote would mitigate that.

Hmm. Actually, I take back what I said in the first sentence. Of the more common terms, maybe distributive is the closest after all. `x * (a + b) = (x * a) + (x * b)`; the left + is `++` and the right + is `>>` and the `x*` is `add $`. If we interpret `+` as `<>` and pretend there's a semigroup instance for `m a`, then that's actually pretty honest.

Reply:
I briefly chatted with Edsko about what name to use for this and although it is an homomorphism from `++` to `>>`, in terms of intuitive wording it is kind of "distributive" that's why I wrote it in quotes.

Let's discuss later what term to use.